### PR TITLE
Try to extract request body payload when the `GetBody` func is nil

### PR DESCRIPTION
**Pull request by tonyredondo**
This `PR` tries to extract request body payload when the `GetBody` func is nil (usually in the request payload from the server side) ~~by using reflection to the inner `LimitedReader`~~

### CODE REVIEW

File: instrumentation/nethttp/client.go [hunk 1]
```diff
@@ -223,16 +225,34 @@ func (t *Transport) doRoundTrip(req *http.Request) (*http.Response, error) {
...
+							bufferSize = size
+						}
+						rqBodyBuffer, err := bufReader.Peek(bufferSize)
+						if err == nil {
+							return string(bytes.Runes(rqBodyBuffer))
+						}
+					}
+				}
+			}
+			return ""
```
Comment (drodriguezhdez) [inline instrumentation/nethttp/client.go]:
```
			if ptr, err := reflection.GetFieldPointerOf(req.Body, "src"); err == nil {
				reader := *(*io.Reader)(ptr)
				if limReader, ok := reader.(*io.LimitedReader); ok {
					if bufReader, ok := limReader.R.(*bufio.Reader); ok {
						size := bufReader.Buffered()
						if size < bufferSize {
							bufferSize = size
						}
						rqBodyBuffer, err := bufReader.Peek(bufferSize)
						if err == nil {
							return string(bytes.Runes(rqBodyBuffer))
						}
					}
				}
			}
			return ""
```

Maybe extract this logic to other function? And make a call like
```
if getBody == nil {
        return getInternalRequestPayload(req.Body)
}
```

Comment (tonyredondo) [reply instrumentation/nethttp/client.go]: I would do that if that code is going to be reused, but right now I don't know what we gain we that.

Comment (drodriguezhdez) [reply instrumentation/nethttp/client.go]: It'd easy to be ready, IMO. I've spent a couple of minutes trying to know where the function returns. 
Additionally, maybe to add a comment explaining when this case is possible (GetBody == null)?

Comment (tonyredondo) [reply instrumentation/nethttp/client.go]: done.

**Pull request merged**